static/js/echo.js

```diff
@@ -447,9 +447,15 @@ $(function () {
     disable_markdown_regex(marked.Lexer.rules.tables, 'heading');
     disable_markdown_regex(marked.Lexer.rules.tables, 'lheading');
 
-    // Disable __strong__, all <em>
+    // Disable __strong__
     marked.InlineLexer.rules.zulip.strong = /^\*\*([\s\S]+?)\*\*(?!\*)/;
-    disable_markdown_regex(marked.InlineLexer.rules.zulip, 'em');
+
+    // Disable _emphasis_
+    // Disable special characters for the emphasis syntax
+    // Inside ** allowed only: word characters/spaces/tabulate
+    // it need for things like "const char *x = (char *)y"
+    marked.InlineLexer.rules.zulip.em = /^\*((?:\*\*|[\w|\ |\t])+?)\*(?!\*)/;
```

Comment:
Why is this regular expression different from the one in bugdown/__init__.py?  It seems like we want to use the same regular expressions in the two places.

Also, why are you limiting the characters inside in any way other than forbidding `*` and `\n` and controlling how we handle whitespace; what I had in mind was just require that the character after the first `*` is a non-whitespace character and the character before the second star is a non-whitespace character.  So `*foo bar*` would be italics but `* foo bar*` and similar things would not be; that covers the `const char *x = (char *)y`  issue fine (since the second `*` has a space in it).